### PR TITLE
Filter unactive communities

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #5470

## Description of Changes
- Adds new query option to get "active" communities (as described by Zak)– those that have at least 1 topic and 10 members
- Adds migration to add `community_id` indexes to Topics and Addresses to potentially speed up query
- Adds caching to the function since the query is still somewhat slow (1.2s) and the data doesn't change frequently

## Test Plan
- Go to http://localhost:8080/communities
  - Confirm that the UI says there are 1K+ communities, but only shows a few hundred

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/communities.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {
   ChainNetwork,
   CommunityCategoryType,
 } from '@hicommonwealth/core';
+import useFetchActiveCommunitiesQuery from 'client/scripts/state/api/communities/fetchActiveCommunities';
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/communities.tsx:6]: I believe it should be staring with `state/`

File: packages/commonwealth/client/scripts/views/pages/communities.tsx [hunk 2]
```diff
@@ -135,7 +136,14 @@ const CommunitiesPage = () => {
     return res;
   };
 
-  const sortedCommunities = sortCommunities(app.config.chains.getAll());
+  const activeCommunities = useFetchActiveCommunitiesQuery();
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/communities.tsx:139]: `  const { data: activeCommunities } = useFetchActiveCommunitiesQuery();`

it's kinda of convention we use. Also, you can add `{ isLoading: isActiveCommunitiesLoading }` if we need spinner

**Comment by jnaviask** [comment]
cc @rbennettcw clarification given

**Comment by rbennettcw** [comment]
Update the query to include communities that are "active" (at least 1 topic, 10 members) *OR* communities that have staking enabled.

**Review by Rotorsoft — APPROVED** [review]
LGTM

**Pull request merged**